### PR TITLE
don't use gnu cp features in building etcd image

### LINKED ISSUE #56383: Hit "cp: illegal option -- t" while making etcd image in macOS
**Is this a BUG REPORT or FEATURE REQUEST?**:

 /kind bug

**What happened**:
hit "cp: illegal option -- t" while making etcd image in macOS

**What you expected to happen**:
etcd image created correctly

**How to reproduce it (as minimally and precisely as possible)**:
```
$ make
# Copy the content in this dir to the temp dir,
# without copying the subdirectories.
find ./ -maxdepth 1 -type f | xargs cp -t /var/folders/mt/xcthpky9237d5drdp2hv76fr0000gn/T/tmp.MnhT8IPQ
cp: illegal option -- t
usage: cp [-R [-H | -L | -P]] [-fi | -n] [-apvXc] source_file target_file
       cp [-R [-H | -L | -P]] [-fi | -n] [-apvXc] source_file ... target_directory
make: *** [build] Error 123
```

**Anything else we need to know?**:

**Environment**: macOS
- Kubernetes version (use `kubectl version`): master branch code
- Cloud provider or hardware configuration:
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`): Darwin Kernel Version 17.2.0: Fri Sep 29 18:27:05 PDT 2017; root:xnu-4570.20.62~3/RELEASE_X86_64 x86_64
- Install tools:
- Others:

### PR DESCRIPTION
**Pull request by dixudx**
**What this PR does / why we need it**:
> cp: illegal option -- t

use gnu-cp to copy files

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #56383

**Special notes for your reviewer**:
/assign @ixdy @mkumatag 

**Release note**:

```release-note
None
```

### CODE REVIEW

File: cluster/images/etcd/Makefile [hunk 1]
```diff
@@ -50,10 +50,20 @@ ifeq ($(ARCH),s390x)
 	BASEIMAGE?=s390x/busybox
 endif
 
+# Use GNU cp
+if LANG=C cp --help 2>&1 | grep -q GNU; then
```
Comment (ixdy) [inline cluster/images/etcd/Makefile]: this is a Makefile, not a shell script, so I don't think these conditionals work?

File: cluster/images/etcd/Makefile [hunk 2]
```diff
@@ -50,10 +50,20 @@ ifeq ($(ARCH),s390x)
...
+else
+    kube::log::error "Failed to find GNU cp as cp or gcp. If you are on Mac: brew install coreutils." >&2
+    return 1
+fi
+
 build:
 	# Copy the content in this dir to the temp dir,
 	# without copying the subdirectories.
-	find ./ -maxdepth 1 -type f | xargs cp -t $(TEMP_DIR)
+	find ./ -maxdepth 1 -type f | xargs ${CP} -t $(TEMP_DIR)
```
Comment (ixdy) [inline cluster/images/etcd/Makefile]: I think the simpler fix would be to replace the right side of this with
`| xargs -I {} cp {} $(TEMP_DIR)`

Comment (dixudx) [reply cluster/images/etcd/Makefile]: @ixdy Nice trick. Thanks.

**Comment by dixudx** [comment]
ping @ixdy @mkumatag PTAL. Needs your `lgtm`. Thanks

**Comment by ixdy** [comment]
(I updated the PR title to more closely match what this is doing now)

**Pull request merged**